signNverify.rst

```diff
@@ -357,3 +375,122 @@ specifying ``--group-id`` can also verify the container:
    3   |1       |NONE    |JSON.Generic
    4   |1       |NONE    |FS
    Container verified: my_container.sif
+
+***********************************
+PEM Key / X.509 Certificate Support
+***********************************
+
+{Project} also supports signing SIF container images
+using a PEM format private key, and verifying with a PEM format public key, or
+X.509 certificate. Non-PGP signatures are implemented using the `Dead Simple
+Signing Envelope <https://github.com/secure-systems-lab/dsse>`__ (DSSE)
+standard.
+
+Signing with a PEM key
+======================
+
+To sign a container using a private key in PEM format, provide the key material
+to the ``sign`` command using the ``--key`` flag:
+
+.. code:: 
+
+   $ {command} sign --key mykey.pem lolcow.sif 
+   INFO:    Signing image with key material from 'mykey.pem'
+   INFO:    Signature created and applied to image 'lolcow.sif'
+
+The DSSE signature descriptor can now be seen by inspecting the SIF file:
+
+.. code:: 
+
+   $ {command} sif list lolcow.sif 
+   ------------------------------------------------------------------------------
+   ID   |GROUP   |LINK    |SIF POSITION (start-end)  |TYPE
+   ------------------------------------------------------------------------------
+   1    |1       |NONE    |32176-32393               |Def.FILE
+   2    |1       |NONE    |32393-33522               |JSON.Generic
+   3    |1       |NONE    |33522-33718               |JSON.Generic
+   4    |1       |NONE    |36864-84656128            |FS (Squashfs/*System/amd64)
+   5    |NONE    |1   (G) |84656128-84658191         |Signature (SHA-256)
+
+   $ {command} sif dump 5 lolcow.sif | jq
+   {
+   "payloadType": "application/vnd.{command}.sif-metadata+json",
+   ...
+
+Attempting to ``verify`` the image without options will fail, as it is not signed with a PGP key:
+
+.. code:: 
+
+   $ {command} verify lolcow.sif 
+   INFO:    Verifying image with PGP key material
+   FATAL:   Failed to verify container: integrity: key material not provided for DSSE envelope signature
+
+Note that the error message shows that the container image has a DSSE signature present.
+
+Verifying with a PEM key
+========================
+
+To verify a container using a PEM public key directly, provide the key material
+to the ``verify`` command using the ``key`` flag:
+
+.. code:: 
+
+   $ {command} verify --key mypublic.pem lolcow.sif 
+   INFO:    Verifying image with key material from 'mypublic.pem'
+   Objects verified:
+   ID  |GROUP   |LINK    |TYPE
+   ------------------------------------------------
+   1   |1       |NONE    |Def.FILE
+   2   |1       |NONE    |JSON.Generic
+   3   |1       |NONE    |JSON.Generic
+   4   |1       |NONE    |FS
+   INFO:    Verified signature(s) from image 'lolcow.sif'
+
+Verifying with an X.509 certificate
```

Comment:
Shouldn't we have a section on signing with certs before there is a section on verifying with certs.  I understand this might be a little harder, but I believe we have added a dummy root, intermediate, and leaf cert [here](https://github.com/apptainer/sif/tree/main/test/keys).  So it should be doable.

Reply:
What do you say we merge this and then you submit another PR with the additions you'd like to see?

Reply:
OK.